accounts-journey/presentation/viewmodel/AccountDetailsViewModel.swift

```diff
@@ -40,9 +46,9 @@ final class AccountDetailsViewModel: NSObject {
             
             switch result {
             case let .success(accountDetailsResponse):
-                screenState = .loaded( accountDetailsResponse.toMapUI())
+                self.screenState = .loaded( accountDetailsResponse.toMapUI())
```

Comment:
Actually using self? is better than the 
           
 ```
guard let self else {
                return
            }
```

Because with this code snippet you create a strong reference to self and it cannot be released until the completion block did not run entirely, while with self? the system will just release it as soon as it needs to.